File: novel-backend/novel-service/src/main/java/com/novel/controller/front/BookController.java
package com.novel.controller.front;

import com.novel.dto.BookChapterDto;
import com.novel.dto.BookInfoDoc;
import com.novel.dto.BookStateVO;
import com.novel.po.book.BookInfo;
import com.novel.result.PageResult;
import com.novel.result.Result;
import com.novel.service.BookService;
import com.novel.service.SearchService;
import com.novel.user.dto.book.*;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.babyfish.jimmer.client.meta.Api;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Api
@Tag(name = "小说")
@RestController
@RequestMapping("/book")
public class BookController {

	@Autowired
	private BookService bookService;
	
	@Autowired
	private SearchService searchService;
	
	@Api
	@Operation(summary = "小说分类查询接口")
	@GetMapping("category")
	public Result<List<BookCategoryView>> listCategory(@Parameter(description = "作品方向 0-男频 1-女频") Integer id) {
		return Result.ok(bookService.listCategory(id));
	}
	
	@Api
	@Operation(summary = "全部小说查询接口")
	@GetMapping("all")
	public Result<PageResult<BookInfoDoc>> listAllCategory(
			AllBookQueryInput params,
			@RequestParam(defaultValue = "1") int pageNum,
			@RequestParam(defaultValue = "20") int pageSize
			) {
		//return Result.ok(bookService.listAll(params, pageNum, pageSize));
		PageResult<BookInfoDoc> searchResult = searchService.conditionSearchBooks(params, pageNum, pageSize);
		return Result.ok(searchResult);
	}
	
	@Api
	@Operation(summary = "小说详情查询接口")
	@GetMapping("info/{id}")
	public Result<BookInfoView> getBookInfo(@PathVariable("id") long id) {
		return Result.ok(bookService.getBookInfo(id));
	}
	
	@Api
	@Operation(summary = "小说目录查询接口")
	@GetMapping("chapter")
	public Result<BookChapterDto> listBookChapter(
			long id,
			@RequestParam(defaultValue = "0") Integer isAll,
			@RequestParam(defaultValue = "1") Integer orderBy) {
		return Result.ok(bookService.listBookChapter(id, isAll, orderBy));
	}
	
	@Api
	@Operation(summary = "随机推荐同类小说")
	@GetMapping("random")
	public Result<List<BookInfo>> listRandomBooks(long categoryId, long bookId) {
		return Result.ok(bookService.listRandomBooks(categoryId, bookId));
	}
	
	@Api
	@Operation(summary = "排行榜单", description = "1-总点击 2-最新入库 3-最新更新 4-评论数")
	@GetMapping("rank")
	public Result<List<BookRankView>> listRankBooks(Integer rankType) {
		return Result.ok(bookService.listRankBooks(rankType));
	}
	
	@Api
	@Operation(summary = "获取章节信息")
	@GetMapping("content")
	public Result<Object> getBookContent(long bookId, long chapterId) {
		BookChapterContentView bookContent = bookService.getBookContent(bookId, chapterId);
		return Result.ok(bookContent);
	}
	
	@Api
	@Operation(summary = "获取小说状态")
	@GetMapping("state")
	public Result<BookStateVO> getBookState(long bookId) {
		return Result.ok(bookService.getBookState(bookId));
	}
	
	@Api
	@Operation(summary = "小说搜索")
	@GetMapping("search")
	public Result<List<BookInfoSearchView>> search(String name) {
		List<BookInfoSearchView> searchResult = searchService.searchBooksByName(name);
		return Result.ok(searchResult);
	}
}

File: frontend-nuxt/app/__generated/services/BookController.ts
import type {Executor} from '../';
import type {Dynamic_BookInfo} from '../model/dynamic/';
import type {
    AllBookQueryInput, 
    BookCategoryView, 
    BookChapterDto, 
    BookInfoDoc, 
    BookInfoSearchView, 
    BookInfoView, 
    BookRankView, 
    BookStateVO, 
    PageResult, 
    Result
} from '../model/static/';

export class BookController {
    
    constructor(private executor: Executor) {}
    
    readonly getBookContent: (options: BookControllerOptions['getBookContent']) => Promise<
        Result<any>
    > = async(options) => {
        let _uri = '/book/content';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.chapterId;
        _uri += _separator
        _uri += 'chapterId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<any>>;
    }
    
    readonly getBookInfo: (options: BookControllerOptions['getBookInfo']) => Promise<
        Result<BookInfoView>
    > = async(options) => {
        let _uri = '/book/info/';
        _uri += encodeURIComponent(options.id);
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<BookInfoView>>;
    }
    
    readonly getBookState: (options: BookControllerOptions['getBookState']) => Promise<
        Result<BookStateVO>
    > = async(options) => {
        let _uri = '/book/state';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<BookStateVO>>;
    }
    
    readonly listAllCategory: (options: BookControllerOptions['listAllCategory']) => Promise<
        Result<PageResult<BookInfoDoc>>
    > = async(options) => {
        let _uri = '/book/all';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.params.bookName;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'bookName='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.channelId;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'channelId='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.categoryId;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'categoryId='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.overState;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'overState='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.wordCountState;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'wordCountState='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.updateDay;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'updateDay='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.params.sortState;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'sortState='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.pageNum;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'pageNum='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.pageSize;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'pageSize='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<PageResult<BookInfoDoc>>>;
    }
    
    readonly listBookChapter: (options: BookControllerOptions['listBookChapter']) => Promise<
        Result<BookChapterDto>
    > = async(options) => {
        let _uri = '/book/chapter';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.id;
        _uri += _separator
        _uri += 'id='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.isAll;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'isAll='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.orderBy;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'orderBy='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<BookChapterDto>>;
    }
    
    readonly listCategory: (options: BookControllerOptions['listCategory']) => Promise<
        Result<ReadonlyArray<BookCategoryView>>
    > = async(options) => {
        let _uri = '/book/category';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.id;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'id='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<BookCategoryView>>>;
    }
    
    readonly listRandomBooks: (options: BookControllerOptions['listRandomBooks']) => Promise<
        Result<ReadonlyArray<Dynamic_BookInfo>>
    > = async(options) => {
        let _uri = '/book/random';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.categoryId;
        _uri += _separator
        _uri += 'categoryId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<Dynamic_BookInfo>>>;
    }
    
    readonly listRankBooks: (options: BookControllerOptions['listRankBooks']) => Promise<
        Result<ReadonlyArray<BookRankView>>
    > = async(options) => {
        let _uri = '/book/rank';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.rankType;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'rankType='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<BookRankView>>>;
    }
    
    readonly search: (options: BookControllerOptions['search']) => Promise<
        Result<ReadonlyArray<BookInfoSearchView>>
    > = async(options) => {
        let _uri = '/book/search';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.name;
        _uri += _separator
        _uri += 'name='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<BookInfoSearchView>>>;
    }
}

export type BookControllerOptions = {
    'listCategory': {
        readonly id?: number | undefined
    }, 
    'listAllCategory': {
        readonly params: AllBookQueryInput, 
        readonly pageNum?: number | undefined, 
        readonly pageSize?: number | undefined
    }, 
    'getBookInfo': {
        readonly id: number
    }, 
    'listBookChapter': {
        readonly id: number, 
        readonly isAll?: number | undefined, 
        readonly orderBy?: number | undefined
    }, 
    'listRandomBooks': {
        readonly categoryId: number, 
        readonly bookId: number
    }, 
    'listRankBooks': {
        readonly rankType?: number | undefined
    }, 
    'getBookContent': {
        readonly bookId: number, 
        readonly chapterId: number
    }, 
    'getBookState': {
        readonly bookId: number
    }, 
    'search': {
        readonly name: string
    }
}
